Sources/PackageManagerDocs/Documentation.docc/Dependencies/AddingSystemLibraryDependency.md

```diff
@@ -0,0 +1,94 @@
+
+### Declaring the system library
+
+The `systemLibrary` definition informs the Swift compiler of where to find the C library.
+When building on Unix-like systems, the package manager can use `pkg-config` to look up where a library is installed.
+Specify the name of the C library you want to look up for the `pkgConfig` parameter.
+When building on Windows, omit `pkgConfig`, as the command `pkg-config` is not expected to be available.
+
+If you omit the optional `pkgConfig` parameter, pass the path of a directory containing the library using the `-L` flag in the command line when building your package.
+For example, if the library you depend on resides in `/usr/local/lib/`, the following command tells the linker to add that path when building the project:
+
+```bash
+% swift build -Xlinker -L/usr/local/lib/
+```
+
+#### System Libraries With Optional Dependencies
+
+To reference a system library with optional dependencies, you need to make another module map package to represent the optional library.
+
+For example, the library `libarchive` optionally depends on `xz`, which means it can be compiled with `xz` support, but it isn't required. 
+To provide a package that uses libarchive with xz, make a `CArchive+CXz` package that depends on `CXz` and provides `CArchive`.
+
+### Defining the module map
+
+The `module.modulemap` file declares the C library headers, and what parts of them, to expose in Swift. 
+A `module.modulemap` file contains one or more maps. 
+Each map defining a name for the Swift module to be exposed, one or more header files to reference, a reference to the name of the C library, and one or more export lines that identify what to expose to Swift.
+
+For example, the following module map uses the header `git2.h`, links to `libgit2`, and exports all functions defined in the header `git2.h` to Swift:
+
+```
+module Clibgit [system] {
+  header "git2.h"
+  link "git2"
+  export *
+}
+```
+
+Module maps must contain absolute paths. As such, they are not cross-platform.
+Use a local header and let the linker handle the location of the binary with a `pkgConfig` parameter in the `systemLibrary` declaration if you can.
+
+For more information on the structure of module maps, see the [LLVM](https://llvm.org/) documentation: [Module Map Language](https://clang.llvm.org/docs/Modules.html#module-map-language).
+
+#### Module Map Versioning
+
+Version the module maps semantically, using your best judgement. 
+Do not follow the version of the system library the module map represents; version the module map(s) independently.
+
+Follow the conventions of system packagers; for example, the Debian package for `python3` is called `python3`. 
+In Debian, there is not a single package for python; the system packagers designed it to be installed side-by-side with other versions. 
+The recommended name for a module map for `python3` on a Debian system is `CPython3`.
```

Comment:
This should refer to the name of the module instead of the name of the module map that defines the module

Reply:
With your feedback and @rauhul - I rewrote this section to focus on the detail of naming based on system conventions, and highlighted the idea that you may want to reserve a module name separate from the C library module interface for an idiomatic Swift wrapper.